subscription.go

```diff
@@ -103,6 +103,12 @@ func (r *marathonClient) registerSubscription() error {
 	case EventsTransportCallback:
 		return r.registerCallbackSubscription()
 	case EventsTransportSSE:
+		if timeout := r.client.config.HTTPClient.Timeout; timeout > 0 {
```

Comment:
go-marathon users don't seem to be able to create 2 different clients right now -- AFAICS, we always reuse a possibly configured custom client for both the regular Marathon API and SSE. Thus, the only option users would have is to not set that timeout at all, which sounds like an overly strong restriction.

Or am I missing something?